clients/drcachesim/tracer/tracer.cpp

```diff
@@ -1466,6 +1466,10 @@ enable_delay_instrumentation()
             event_delay_bb_analysis, event_delay_app_instruction, &memtrace_pri))
         DR_ASSERT(false);
     schedule_tracing_lock = dr_mutex_create();
+#if defined(AARCH64) && defined(DELAYED_CHECK_INLINED)
+    /* By counting down we can avoid clobbering aflags in our comparison. */
+    instr_count = op_trace_after_instrs.get_value();
```

Comment:
This will hit the threshold after op_trace_after_instrs.get_value() + 1 instructions, since it calls the hit_instr_count_threshold when the counter becomes negative (top bit turns to 1). If I am not missing something, this should change to instr_count = op_trace_after_instrs.get_value() - 1;

Reply:
Right, changed to -1.  There is no test of the precise count -- not sure it's worth trying to make one, since this isn't used as a precise trigger in tracing use cases.

Reply:
Yeah I do not think it is worth making a test for precise count. For practical purposes even the one-off does not matter but it was worth fixing since I noticed it.